### PR TITLE
[Feature] Support TTA for Segmentor

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support TTA for segmentor

documentation: https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/test_time_augmentation.html
Related PR: https://github.com/open-mmlab/mmdetection/pull/9452 & https://github.com/open-mmlab/mmsegmentation/pull/2184/

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/_base_/datasets/scannet-seg.py [hunk 1]
```diff
@@ -109,6 +90,48 @@
     dict(type='NormalizePointsColor', color_mean=None),
     dict(type='Pack3DDetInputs', keys=['points'])
 ]
+tta_pipeline = [
```
[the remark this answers is not in the document]
Comment (Xiangxu-0103) [reply configs/_base_/datasets/scannet-seg.py:93]: In the newest version, we need to use `tta_pipeline` to support TTA model, more detailed information can be found in [MMEngine](https://mmengine.readthedocs.io/zh_CN/latest/advanced_tutorials/test_time_augmentation.html), and related PR in [mmdet](https://github.com/open-mmlab/mmdetection3d/pull/2382) and [mmseg](https://github.com/open-mmlab/mmsegmentation/pull/2184).

File: configs/_base_/datasets/semantickitti.py [hunk 2]
```diff
@@ -112,12 +112,21 @@
         seg_offset=2**16,
         dataset_type='semantickitti',
         backend_args=backend_args),
-    dict(type='PointSegClassMapping', ),
-    dict(type='Pack3DDetInputs', keys=['points', 'pts_semantic_mask'])
```
[the remark this answers is not in the document]
Comment (Xiangxu-0103) [reply configs/_base_/datasets/semantickitti.py:116]: For test dataset, we do not need to output annotations, but use groundtruth from `eval_ann_info` generated from `PointSegClassMapping`.

File: tests/test_datasets/test_tta.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+class TestMuitiScaleFlipAug3D(TestCase):
+
+    def test_exception(self):
+        with pytest.raises(TypeError):
+            tta_transform = dict(
+                type='TestTimeAug',
+                transforms=[
+                    dict(
+                        type='mmdet3d.RandomFlip3D',
```
Comment (ZCMax) [inline tests/test_datasets/test_tta.py:20]: why do we `mmdet3d.RandomFlip3D` instead of `RandomFlip3D`?

File: tools/test.py [hunk 4]
```diff
@@ -109,6 +111,12 @@ def main():
     if args.show or args.show_dir:
         cfg = trigger_visualization_hook(cfg, args)
 
+    if args.tta:
```
Comment (ZCMax) [inline tools/test.py:114]: Please notify that currently we only support 3D Segmentation TTA

**Review by sunjiahao1999 — APPROVED** [review]

**Pull request merged**